### PR TITLE
Fix UI for select property in view headers menu

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Discussion](https://discord.com/channels/894960387743698944/1037760892965748827/threads/1192592148793655317)

### CODE REVIEW

File: components/common/BoardEditor/focalboard/src/components/viewHeader/viewHeaderRowsMenu/PersonPropertyTemplateMenu.tsx [hunk 1]
```diff
@@ -1,63 +1,24 @@
...
-                    },
-                    scope: WebhookEventNames.CardPersonPropertyAssigned
-                  } as CreateEventPayload)
-              )
-            )
-            .flat()
-        });
-      }
-    },
+    onChange,
```
Comment (Devorein) [inline components/common/BoardEditor/focalboard/src/components/viewHeader/viewHeaderRowsMenu/PersonPropertyTemplateMenu.tsx:21]: Updating person property from view header menu was broken before.

File: components/common/BoardEditor/focalboard/src/components/viewHeader/viewHeaderRowsMenu/PropertyTemplateMenu.tsx [hunk 2]
```diff
@@ -185,7 +201,7 @@ export function PropertyTemplateMenu({
     }
 
     default: {
-      return null;
+      return <TextPropertyTemplateMenu onChange={onChange} cards={checkedCards} propertyTemplate={propertyTemplate} />;
```
Comment (Devorein) [inline components/common/BoardEditor/focalboard/src/components/viewHeader/viewHeaderRowsMenu/PropertyTemplateMenu.tsx:204]: Another bug that was fixed. Text-based properties were not showing up on view header menu.

File: components/common/CharmEditor/components/BlockAligner.tsx [hunk 3]
```diff
@@ -40,7 +40,7 @@ const Controls = styled.div`
   transition: opacity 250ms ease-in-out;
 `;
 
-const BlockAligner = forwardRef<HTMLDivElement, BlockAlignerProps>((props) => {
+const BlockAligner = forwardRef<HTMLDivElement, BlockAlignerProps>((props, ref) => {
```
Comment (Devorein) [inline components/common/CharmEditor/components/BlockAligner.tsx:43]: Changed to remove ref warning

**Review by mattcasey — APPROVED** [review]

File: components/proposals/components/NewProposalButton.tsx [hunk 4]
```diff
@@ -55,7 +55,8 @@ export function NewProposalButton() {
   const canCreateProposal = spacePermissions?.createProposals;
   // grab page data from context so that title is always up-to-date
   const proposalTemplatePages = proposalTemplates
-    ?.map(
+    ?.filter((template) => !!pages[template.page.id])
```
Comment (mattcasey) [inline components/proposals/components/NewProposalButton.tsx:58]: curious if there was an issue that this caused?

Comment (Devorein) [reply components/proposals/components/NewProposalButton.tsx:58]: It was due to a unique key constraint issue, the proposal data weren't present in the object. Thus there was no id property leading to undefined being passed to the key

**Pull request merged**